tasks/update_go.py

```diff
@@ -0,0 +1,179 @@
+        )
+
+    with open(path, "w", newline='') as writer:
+        writer.write(content)
+
+
+# returns the current go version
+def _get_repo_go_version() -> str:
+    with open(GO_VERSION_FILE, "r") as reader:
+        version = reader.read()
+    return version.strip()
+
+
+# extracts the major version from the given string
+# eg. if the string is "1.2.3", returns "1.2"
+def _get_major_version(version: str) -> str:
+    major, _, _ = version.rpartition(".")
+    return major
+
+
+def _update_go_version_file(version: str):
+    _update_file(GO_VERSION_FILE, "[.0-9]+", version)
+
+
+def _update_gdb_dockerfile(version: str):
+    path = "./tools/gdb/Dockerfile"
+    pattern = r'(https://go\.dev/dl/go)[.0-9]+(\.linux-amd64\.tar\.gz)'
+    replace = rf'\g<1>{version}\g<2>'
+    _update_file(path, pattern, replace)
+
+
+def _update_install_devenv(version: str):
+    path = "./devenv/scripts/Install-DevEnv.ps1"
+    _update_file(path, '("Installing go )[.0-9]+"', rf'\g<1>{version}"')
+    _update_file(
+        path,
+        r'(https://dl\.google\.com/go/go)[.0-9]+(\.windows-)',
+        rf'\g<1>{version}\g<2>',
+        2,
+    )
+
+
+def _update_agent_devenv(version: str):
+    path = "./docs/dev/agent_dev_env.md"
+    pattern = r"^(You must \[install Golang\]\(https://golang\.org/doc/install\) version )`[.0-9]+`"
+    replace = rf"\g<1>`{version}`"
+    _update_file(path, pattern, replace)
+
+
+def _update_task_go(version: str):
+    path = "./tasks/go.py"
+    pattern = '("go version go)[.0-9]+( linux/amd64")'
+    replace = rf'\g<1>{version}\g<2>'
+    _update_file(path, pattern, replace)
+
+
+def _update_readme(major: str):
+    path = "./README.md"
+    pattern = r'(\[Go\]\(https://golang\.org/doc/install\) )[.0-9]+( or later)'
+    replace = rf'\g<1>{major}\g<2>'
+    _update_file(path, pattern, replace)
+
+
+def _update_go_mods(major: str):
+    mod_files = [f"./{module}/go.mod" for module in DEFAULT_MODULES]
+    for mod_file in mod_files:
+        _update_file(mod_file, "^go [.0-9]+$", f"go {major}")
+
+
+def _create_releasenote(ctx: Context, version: str):
+    RELEASENOTE_TEMPLATE = """---
+enhancements:
+- |
+    Agents are now built with Go ``{}``.
+"""
+    # hiding stderr too because `reno` displays some warnings about the config
+    res = ctx.run(f'reno new "bump go to {version}"', hide='both')
+    match = re.match("^Created new notes file in (.*)$", res.stdout, flags=re.MULTILINE)
+    if not match:
+        raise exceptions.Exit("Could not get created release note path")
+
+    path = match.group(1)
+    with open(path, "w") as writer:
+        writer.write(RELEASENOTE_TEMPLATE.format(version))
+    return path
+
+
+@task(
+    help={
+        "version": "The version of Go to use",
+        "image_tag": "Tag from buildimages with format v<build_id>_<commit_id>",
+        "test_version": "Whether the image is a test image or not",
+        "force_major": "Whether to apply changes like if it was a major version update",
```

Comment:
Can you explain what is the use case of this flag?

Reply:
The set of changes we have to do is not the same if it's a minor or a major update (a few files only mention the "major" part of the version, ie. `1.20`).
To avoid trying to change everything for a minor update, I put these extra changes in an `if` block, this flag forces performing the changes even for a minor update.
I guess we could just always do them to make sure every file is up-to-date anyway, what do you think ?

Reply:
Removed the `force_major` flag, so just doing everything no matter if it's a "minor" or a "major" version update (in case of "minor" update then those files shouldn't change anyway)